src/page_action/Components/App.svelte

```diff
@@ -0,0 +1,71 @@
+<script>
+  import { onMount } from "svelte";
+
+  const ACCESS_TOKEN_KEY = "accessToken";
+  const GOOGLE_REVOKE_TOKEN_URL = "https://accounts.google.com/o/oauth2/revoke";
+
+  let accessToken = "";
+  let getAuthTokenPromise;
+
+  function handleLogin(e) {
+    getAuthTokenPromise = getAuthToken();
+  }
+
+  function handleRevoke(e) {
+    revokeToken();
+  }
+
+  const revokeToken = async function revokeToken() {
+    await fetch(`${GOOGLE_REVOKE_TOKEN_URL}?token=${accessToken}`);
+    await removeCachedAuthToken();
+    await removeAccessTokenFromStorage();
+  };
+
+  function removeCachedAuthToken() {
+    return new Promise(function(resolve) {
+      chrome.identity.removeCachedAuthToken({ token: accessToken }, function() {
+        accessToken = "";
+        resolve(true);
+      });
+    });
+  }
+
+  function removeAccessTokenFromStorage(params) {
+    return new Promise(function(resolve) {
+      chrome.storage.local.remove([ACCESS_TOKEN_KEY], function() {
+        console.log("accessToken removed from storage");
+        resolve(true);
+      });
+    });
+  }
+
+  function getAuthToken() {
+    return new Promise(function(resolve) {
+      chrome.identity.getAuthToken({ interactive: true }, function(token) {
+        chrome.storage.local.set({ [ACCESS_TOKEN_KEY]: token }, function() {
+          accessToken = token;
+          resolve(accessToken);
+        });
+      });
+    });
+  }
+
+  onMount(() => {
+    chrome.storage.local.get([ACCESS_TOKEN_KEY], function(result) {
+      if (result.accessToken) {
```

Comment:
```suggestion
      accessToken = result.accessToken;
      if (result.accessToken) {
```

One more line to avoid interface blinking. May be helpful